tests/test_updated_meta.py

```diff
@@ -0,0 +1,117 @@
+CROP_FOREGROUND = mt.CropForegroundd(IM, IM)
+MAX_DIFF = 3  # max allowed difference (both voxels and mm)
+
+TESTS = []
+for trans in (SPATIAL_CROP, CROP_FOREGROUND):
+    for resample in (False, True):
+        TESTS.append((trans, resample))
+
+
+class TestUpdatedMeta(unittest.TestCase):
+    def setUp(self):
+        affine = make_rand_affine()
+        _, im = create_test_image_3d(*IM_SHAPE)
+        monai_data = os.environ.get("MONAI_DATA_DIRECTORY") or tempfile.mkdtemp()
+        self.folder = os.path.join(monai_data, "test_updated_meta")
+        self.im_fname = make_nifti_image(im, affine, dir=self.folder, fname="orig", verbose=True)
+        self.loader = mt.Compose([mt.LoadImaged(IM), mt.AddChanneld(IM)])
+
+    @staticmethod
+    def get_center_mass(data, key, select_fn=lambda x: x > 0):
+        """
+        Get center of mass of all voxels that conform to select_fn.
+        Return COM in both image and real space.
+        """
+        img = data[key]
+        if img.ndim != 4 or img.shape[0] != 1:
+            raise RuntimeError("Expect shape to be `CHWD` where `C==1`.")
+        img = select_fn(img[0])  # remove channel dim and convert to binary mask
+        affine = data[PostFix.meta(key)]["affine"]
+        com_img = [np.average(i) for i in np.where(img)]
+        com_real = affine @ (com_img + [1])
+        com_real = com_real[:-1]
+        return np.asarray(com_img), com_real
+
+    @staticmethod
+    def get_diff_com(com1, com2):
+        """Euclidean distance between two COMs."""
+        return np.sum((com1 - com2) ** 2) ** 0.5
+
+    def check_coms(self, d1, d2, img_should_match=False):
+        """
+        Check that COMs match in real space.
+        If image has not been resampled, there should be a discrepency in image space.
+        """
+        com1_img, com1_real = self.get_center_mass(d1, IM)
+        com2_img, com2_real = self.get_center_mass(d2, IM)
+        com_diff_img = self.get_diff_com(com1_img, com2_img)
+        com_diff_real = self.get_diff_com(com1_real, com2_real)
+
+        # if images have been resampled, then they should match in image space. else, not.
+        if img_should_match:
+            self.assertLess(com_diff_img, MAX_DIFF)
+        else:
+            self.assertGreater(com_diff_img, MAX_DIFF)
+        # either way, check they match in real space.
+        self.assertLess(com_diff_real, MAX_DIFF)
+
+    @parameterized.expand(TESTS)
+    def test_updated_meta(self, extra_trans, resample):
+        d_in = self.loader({IM: self.im_fname})
+        d_out = extra_trans(deepcopy(d_in))
+        self.check_coms(d_in, d_out)
+
+        # check that saving to file has the same result
+        prefix = "spatial_crop" if isinstance(extra_trans, mt.SpatialCropd) else "crop_foreground"
+        if resample:
+            prefix += "_resample"
+
+        # TODO: remove this line
+        if resample:
+            del d_out[PostFix.meta(IM)]["original_affine"]
```

Comment:
This part is worrying. If `resample==True`, then a comparison is done between `original_affine` and `affine`. Since `affine` has been modified compared to the current MONAI implementation, this causes double resampling.
Any ideas how to remove this?

Reply:
how about replacing line 103  with `resample=False` which turns off the resample?